gem/backends.py

```diff
@@ -36,15 +37,34 @@ def _update_user_from_claims(user, claims):
     user.email = claims.get("email", "")
     user.save()
 
+    username = claims.get("username", "")
+
     # If the user doesn't have a profile for some reason make one
     if not hasattr(user, 'profile'):
         user.profile = UserProfile(user=user)
+
+        # TODO: we should be using a more specific site here?
         user.profile.site = Site.objects.get(is_default_site=True)
 
     # Ensure the profile is linked to their auth service account using the uuid
     if user.profile.auth_service_uuid is None:
         user.profile.auth_service_uuid = claims.get("sub")
 
+        # If a user already exists with this username
+        # change that user's username
+        if username:
+            for u in User.objects.filter(
+                    username=username).exclude(pk=user.pk):
+                if user.profile.auth_service_uuid is None:
+                    user.username = str(user.profile.site.pk) + '_' + username
+                    user.save()
```

Comment:
You're using the wrong variable throughout here. `u` is the user that we're clashing with, we want to be changing their details. Right now you're checking and changing the details for the person logging in.